plugins/database/postgresql/postgresql.go

```diff
@@ -35,6 +37,26 @@ ALTER ROLE "{{username}}" WITH PASSWORD '{{password}}';
 	defaultUserNameTemplate = `{{ printf "v-%s-%s-%s-%s" (.DisplayName | truncate 8) (.RoleName | truncate 8) (random 20) (unix_time) | truncate 63 }}`
 )
 
+type PasswordEncryption string
+
+var (
+	PasswordEncryptionSCRAMSHA256 PasswordEncryption = "scram-sha-256"
+	PasswordEncryptionPlainText   PasswordEncryption = "plaintext"
+)
+
+var passwordEncryptions = map[PasswordEncryption]struct{}{
+	PasswordEncryptionSCRAMSHA256: {},
+	PasswordEncryptionPlainText:   {},
+}
+
+func parsePasswordEncryption(s string) (PasswordEncryption, error) {
+	if _, ok := passwordEncryptions[PasswordEncryption(s)]; !ok {
+		return "", fmt.Errorf("'%s' is not a valid password encryption type", s)
+	}
+
+	return PasswordEncryption(s), nil
+}
```

Comment:
Nit: should we put the password encryption stuff in its own file since postgresql.go is getting long-ish?
Enum-style custom types are usually well scoped and good candidates to scope out so the main file is more readable.

Reply:
moved to its own file